rfcs/0003-storage.md

```diff
@@ -1,46 +1,108 @@
 
-# Network
+# 4. Network
 
 Validators for PoRep are the same validators that are verifying transactions. They have some stake that they have put up as collateral that ensures that their work is honest. If you can prove that a validator verified a fake PoRep, then the validators stake can be slashed.
 
 Replicators are specialized thin clients. They download a part of the ledger and store it, and provide PoReps of storing the ledger. For each verified PoRep replicators earn a reward of sol from the mining pool.
 
-# Constraints
+# 5. Constraints
 
 We have the following constraints:
-* At most 14k  replication identities can be used, because thats how many CUDA cores we can fit in a $5k box at the moment.
-* Verification requires generating the CBC blocks. That requires space of 2 blocks per identity, and 1 CUDA core per identity for the same dataset. So as many identities at once should be batched with as many proofs for those identities verified concurrently for the same dataset.
-
-# Validation and Replication Protocol
-
-1. The network sets a replication target number, let's say 1k. 1k PoRep identities are created from signatures of a PoH hash. So they are tied to a specific PoH hash. It doesn't matter who creates them, or it could simply be the last 1k validation signatures we saw for the ledger at that count. This is maybe just the initial batch of identities, because we want to stagger identity rotation.
-2. Any client can use any of these identities to create PoRep proofs. Replicator identities are the CBC encryption keys.
-3. Periodically at a specific PoH count, a replicator that wants to create PoRep proofs signs the PoH hash at that count. That signature is the seed used to pick the block and identity to replicate. A block is 1TB of ledger.
-4. Periodically at a specific PoH count, a replicator submits PoRep proofs for their selected block. A signature of the PoH hash at that count is the seed used to sample the 1TB encrypted block, and hash it. This is done faster than it takes to encrypt the 1TB block with the original identity.
-5. Replicators must submit some number of fake proofs, which they can prove to be fake by providing the seed for the hash result.
-6. Periodically at a specific PoH count, validators sign the hash and use the signature to select the 1TB block that they need to validate. They batch all the identities and proofs and submit approval for all the verified ones.
-7. After #6, replicator client submit the proofs of fake proofs.
+*  Verification requires generating the CBC blocks. That requires space of 2 blocks per identity, and 1 CUDA core per identity for the same dataset. So as many identities at once should be batched with as many proofs for those identities verified concurrently for the same dataset.
+* Validators will randomly sample the set of storage proofs to the set that they can handle, and only the creators of those chosen proofs will be rewarded.
```

Comment:
how do they determine how many they can handle?

Reply:
Probably something like a benchmark that gets run on the start of the mining program which gives an estimate of how many samples they can process in a given period.